doc/Language/filename-extensions.rakudoc

```diff
@@ -0,0 +1,49 @@
+=begin pod :kind("Language") :subkind("Language") :category("beginning")
+
+=TITLE Filename extensions
+
+=SUBTITLE The extensions recommended for files with Raku content.
+
+=begin table
+File contents      | Extensions recommended or used at one time | Current recommendation
+===================================================================================
+B<Language files>  |                                            |
+Raku script        | .pl, .p6                                   | .raku
+Raku module        | .pm, .pm6                                  | .rakumod
+Raku documentation | .pm, pm6, pod, pod6                        | .rakudoc
+Test files in raku | .t                                         | .rakutest
+NotQuitePerl or NQP |                                           | .nqp
+B<Module specific files> |                                      |
+Cro templates      |                                            | .crotmp
+
+=end table
+
+=head1 Introduction
+
+A filename extension is defined, at least for this page, as the remaining letters after and including
+the final '.'. A filename without a '.' in the filename does not have a filename extension.
+
+Filename extensions can be used to associate behaviour with the file content, such as opening
+a specialised reader or editor, running a program with the file content as input, etc. On some
+operating systems, this is almost the required behaviour. Even when it is not required by the
+operating system, filename extensions provide a useful mnenomic, or reminder, about the contents
+of the file.
+
+Early filesystems, the part of the operating system handling filenames, constrained filename extensions
+to three characters after the '.'. This is no longer a constraint.
+
+=head1 History and Extensions
+
+The Raku language was for many years known as Perl 6, and filename extensions
+were adopted accordingly. Consequently, there is a best current practice and a previous best
+practice. The following table is an attempt to list all the filename extensions, starting with the
+earliest on the left.
+
+The most recent best practice is shown in the last column.
+
+Some modules will become widely used, and will have filename extensions associated with them.
+
+A module developer / maintainer may add to the table and create a PR with the change in the
```

Comment:
This is probably redundant.

Reply:
Redundant yes, but I tend to restate things for beginners